book/api/websocket.md

```diff
@@ -0,0 +1,1093 @@
+| waterfall        | `TxnWaterfall` | A waterfall of transactions received since the end of the previous leader slot |
+
+A transaction waterfall describes the transactions that are received
+before and during a leader slot, and what happened to them. A typical
+waterfall is that we acquire transactions from QUIC or gossip in the
+lead up to (before) our leader slot, drop a few of them that fail to
+verify, drop a few duplicates, drop some low priority ones that won't
+fit into our block, and then successfully place some transactions into
+a block. Transactions can also be received and dropped during the leader
+slot, but it's important to note: the waterfall shows statistics for all
+transactions since the end of our last leader slot. These are
+transactions that are now eligible for palcement into the next one.
+
+The waterfall is typically useful when viewing what happened in a past
+leader slot: we want to know where transactions came from, and for what
+reasons they didn't make it into the block. For example, if we received
+100,000 transactions leading up to the slot, but only 6000 made it in,
+what happened to the other 94,000?
+
+The live waterfall is a special case: it's for the next slot of the
+validator, rather than one that is in the past. Becuase the slot hasn't
+happened yet, we know certain information: how many transactions we have
+received so far from users that we could pack into our next block, how
+many have expired, how many failed to verify, and so on, but we probably
+won't know how many made it into the block yet, as we do when looking at
+the waterfall for a block that has been published.
+
+The waterfall should generally be balanced: total transactions in and
+total transactions out will be the roughly the same, but not always
+strictly. Transactions in could be more or less than transactions out
+due to sampling jiter. When subtracting, be sure to account for
+potential underflow.
+
+#### `summary.live_tile_primary_metric`
+| frequency        | type                    | example |
+|------------------|-------------------------|---------|
+| *Once* + *100ms* | `LiveTilePrimaryMetric` | below   |
+
+::: details Example
+
+```json
+{
+    "topic": "summary",
+    "key": "live_tile_primary_metric",
+    "value": {
+        "next_leader_slot": 285228774,
+        "tile_primary_metric": {
+            "quic": 3,
+            "net_in": 37803082,
+            "net_out": 4982399,
+            "verify": 0,
+            "dedup": 0,
+            "bank": 89407,
+            "pack": 0,
+            "poh": 0,
+            "shred": 0,
+            "store": 0
+        }
+    }
+}
+```
+
+:::
+
+**`LiveTilePrimaryMetric`**
+| Field               | Type                | Description
+|---------------------|---------------------|------------
+| next_leader_slot    | `number\|null`      | The next leader slot |
+| tile_primary_metric | `TilePrimaryMetric` | Per-tile-type primary metrics since the end of the previous leader slot |
```

Comment:
These are averaged since the end of the previous leader slot?

Reply:
Some are.  Others are more point-in-time values in nature.  Will clarify.